### PR TITLE
feat: Buttons

### PR DESCRIPTION
**Pull request by frassinier**
buttons

### CODE REVIEW

**Comment by acateland** [comment]
Looks like you prepared the work for other components seeing the variable file

File: src/theme/_buttons.scss [hunk 1]
```diff
@@ -1,14 +1,51 @@
...
-$btn-inverse-color: White !default;
-$btn-inverse-border: transparent !default;
+//
+// Buttons
+// --------------------------------------------------
 
-$btn-add-bg: #bad104 !default;
-$btn-add-color: White !default;
-$btn-add-border: darken($btn-add-bg, 5%) !default;
+$btn-box-shadow-color: 20 !default;
```
Comment (frassinier) [inline src/theme/_buttons.scss]: This variables has a wrong name. I will fix it!

**Comment by frassinier** [comment]
Yes, indeed, I will prepare vars to remove colors from all components.

File: src/theme/_helpers.scss [hunk 2]
```diff
@@ -0,0 +1,20 @@
+/// Some utils functions
+/// @source https://css-tricks.com/snippets/sass/tint-shade-functions/
+
+/// Slightly lighten a color
+/// @access public
+/// @param {Color} $color - color to tint
+/// @param {Number} $percentage - percentage of `$color` in returned color
+/// @return {Color}
```
Comment (talend-glorieux) [inline src/theme/_helpers.scss]: What is the documentation format used here?
Meaning: Can we actually extract it to generate a documentation or is it only here to enhance code comprehension.

File: src/theme/_buttons.scss [hunk 3]
```diff
@@ -1,14 +1,51 @@
-$btn-inverse-bg: transparent !default;
-$btn-inverse-color: White !default;
-$btn-inverse-border: transparent !default;
+//
+// Buttons
+// --------------------------------------------------
```
Comment (talend-glorieux) [inline src/theme/_buttons.scss]: I really have issues with gigantic file headers / ascii art:

* I don't feel like they improve the file comprehension. 
* The file name is already here for that.
* It's one more thing to maintain when you're changing file names.  

Not sure if it's only me. 
@Talend/frontend Any thoughts on it ?

Comment (acateland) [inline src/theme/_buttons.scss]: agree, except if it host some documentation

File: src/theme/_helpers.scss (hunk 2 above)
Comment (frassinier) [inline src/theme/_helpers.scss]: Take a look at http://sassdoc.com/

File: src/theme/_buttons.scss (hunk 3 above)
Comment (frassinier) [inline src/theme/_buttons.scss]: I just fork bootstrap-sass one :simple_smile:

**Comment by talend-sromain** [comment]
lgtm

File: src/theme/_buttons.scss [hunk 4]
```diff
@@ -1,14 +1,52 @@
...
-$btn-add-border: darken($btn-add-bg, 5%) !default;
+$btn-box-shadow-width:          3px !default;
+$btn-box-shadow-shade-percent:  20 !default;
 
-.btn-inverse {
-  @include button-variant($btn-inverse-color, $btn-inverse-bg, $btn-inverse-border);
+/// Set box shadow to a button
+/// @access public
+/// @param {Color} $border-bottom-color - color to shade
+@mixin btn-box-shadow($btn-bg: $btn-default-bg) {
```
Comment (acateland) [inline src/theme/_buttons.scss]: could it be made more generic, i mean it could be and may be used on more components thant buttons  ?

Comment (frassinier) [inline src/theme/_buttons.scss]: I propose to let that piece of code here and when - if needed - we need to make it more generic we try to abstract it.  
For all UI assets that I have seen, only buttons have a box shadow.

**Comment by acateland** [comment]
lgtm

**Pull request merged**